cmd/earthly/subcmd/build_cmd.go

```diff
@@ -500,21 +500,28 @@ func (a *Build) ActionBuildImp(cliCtx *cli.Context, flagArgs, nonFlagArgs []stri
 	}
 
 	cacheImports := make([]string, 0)
+	var cacheImportImageName string
 	if a.cli.Flags().RemoteCache != "" {
-		cacheImports = append(cacheImports, a.cli.Flags().RemoteCache)
+		cacheImportImageName, _, err = a.parseImageNameAndAttrs(a.cli.Flags().RemoteCache)
+		cacheImports = append(cacheImports, cacheImportImageName)
 	}
 	if len(a.cacheFrom.Value()) > 0 {
 		cacheImports = append(cacheImports, a.cacheFrom.Value()...)
 	}
 	var cacheExport string
 	var maxCacheExport string
+	var cacheExportAttrs map[string]string
 	if a.cli.Flags().RemoteCache != "" && a.cli.Flags().Push {
 		if a.cli.Flags().MaxRemoteCache {
-			maxCacheExport = a.cli.Flags().RemoteCache
+			maxCacheExport, cacheExportAttrs, err = a.parseImageNameAndAttrs(a.cli.Flags().RemoteCache)
 		} else {
-			cacheExport = a.cli.Flags().RemoteCache
+			cacheExport, cacheExportAttrs, err = a.parseImageNameAndAttrs(a.cli.Flags().RemoteCache)
 		}
 	}
+	if err != nil {
```

Comment:
can you do this immediately after each call

and please use a clearer error message, like the one I suggested earlier:

> return nil, errors.Wrapf(err, "parse max cache export %s", s.maxCacheExport)

Reply:
this hasn't been fully addressed -- can you do this immediately after each call

Reply:
I've added it in one more place